### PR TITLE
asserts: introduce snap-declaration

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
requires #493

File: asserts/snap_asserts_test.go [hunk 1]
```diff
@@ -29,29 +29,98 @@ import (
...
+		"snap-id: snap-id-1\n" +
+		"snap-name: first\n" +
+		"publisher-id: dev-id1\n" +
+		"gates: snap-id-3,snap-id-4\n" +
+		sds.tsLine +
+		"body-length: 0" +
+		"\n\n" +
+		"openpgp c2ln"
+
+	invalidTests := []struct{ original, invalid, expectedErr string }{
```
Comment (emgee) [inline asserts/snap_asserts_test.go]: It's missing a test for the `series` header field I think.

File: asserts/snap_asserts_test.go [hunk 2]
```diff
@@ -29,29 +29,98 @@ import (
...
+		{"snap-id: snap-id-1\n", "", `"snap-id" header is mandatory`},
+		{"snap-name: first\n", "", `"snap-name" header is mandatory`},
+		{"publisher-id: dev-id1\n", "", `"publisher-id" header is mandatory`},
+		{"gates: snap-id-3,snap-id-4\n", "gates: foo,\n", `empty entry in comma separated "gates" header: "foo,"`},
```
Comment (emgee) [inline asserts/snap_asserts_test.go]: Test an empty `gates` header field too?

**Comment by mvo5** [comment]
Looks good, +1 from me once the points that Matt raised are addressed.

**Comment by mvo5** [comment]
retest this please

**Comment by pedronis** [comment]
retest this please

File: asserts/snap_asserts_test.go (hunk 1 above)
Comment (pedronis) [inline asserts/snap_asserts_test.go]: that one is part of the primary key so checking for those is not dependent on type specific code anymore, added a test anyway for completeness

**Pull request merged**